src/locations-detailed.md

```diff
@@ -19,18 +24,70 @@ The second two-word special case is that of pointers to calldata lookup types;
 see the section on [pointers to calldata from the
 stack](#user-content-locations-in-detail-calldata-in-detail-pointers-to-calldata-pointers-to-calldata-from-the-stack) for details.
 
-The stack is a bit unpredictable in terms of data layout, as it's also used as
-working space.  However, the location of local variables can be figured out by
-other parts of the debugger, so we won't go into it here.  The location of
-function parameters can also be figured out by the debugger, but we need to
-discuss here the order in which such parameters go on the stack, so let us do
-that.
-
-Function parameters go on the bottom of a function's stackframe, directly above
-the return address (if there is one -- constructors don't have them).  They go
-in the order of first input parameters, in the order they were given, followed
-by output parameters, in the order they were given.  Anonymous output parameters
-are treated the same as named output parameters for these purposes.
+#### The stack: Data layout
+
+Stack variables are local variables, so naturally things will change as the
+contract executes.  But, we can still describe how things are at any given
+time.  Note that if you are actually writing a debugger, you may want to rely
+on other systems to determine data layout on the stack.
+
+The stack is of course not used only for storing local variables, but also as a
+working space.  And of course it also holds return addresses.  The stack is
+divided into stackframes; each stackframe begins with the return address.
+(There is no frame pointer, for those used to such a thing; just a return
+address.)  The exception is constructors, which do not include a return
+address.  In addition, if the initial function call (i.e. stackframe) of the
+EVM stackframe (i.e. message call or creation call) is not a constructor or
+fallback function, the function selector will be stored on the stack below the
+first stackframe.
+
+Note that only function calls create new stackframes; function modifiers and
+base constructor invocations (whether placed on the constructor or on the
+contract) do not create new stackframes.
+
+Within each stackframe, all variables are always stored below the workspace. So
+while the workspace may be unpredictable, we can ignore it for the purposes of
+data layout within a given stackframe.  (Of course, the workspace in one
+stackframe does come between that stackframe's variables and the start of the
+next stackframe.)
+
+Restricting our attention the variables, then, the stack acts, as expected, as
+a stack; variables are pushed onto it when needed, and are popped off of it
```

Comment:
`:` seems to make more sense here than `;`

Reply:
Leaving this one as-is, I think